subprojects/docs/docs.gradle

```diff
@@ -92,14 +86,14 @@ gradlebuildJava {
 ext {
     srcDocsDir = file('src/docs')
     releaseFeaturesFile = file("$srcDocsDir/release/release-features.txt")
+    docsDir = new File(buildDir, 'docs')
     userguideSrcDir = new File(srcDocsDir, 'userguide')
-    asciidocSanitizeStylesheet = file("$srcDocsDir/stylesheets/asciidocSanitize.xsl")
-    asciidocOutputDir = new File(buildDir, 'asciidoc/userguide')
     dslSrcDir = new File(srcDocsDir, 'dsl')
-    docsDir = file("$buildDir/docs")
-    userguideDir = new File(docsDir, 'userguide')
+    userguideIntermediateOutputDir = new File(buildDir, 'userguideIntermediate')
+    userguidePdfOutputDir = new File(buildDir, 'userguidePdf')
+    userguideOutputDir = new File(docsDir, 'userguide')
     distDocsDir = new File(buildDir, 'distDocs')
-    samplesDir = file("$buildDir/samples")
+    samplesDir = new File(buildDir, 'samples')
     docbookSrc = new File(project.buildDir, 'src')
     samplesSrcDir = file('src/samples')
```

Comment:
Why `new File(...)` rather than `file(...)`? I don't like to encourage folks to use the former because they often aren't aware of the implications. The only reason it works here is because `buildDir` is an absolute path.

Reply:
Also, can we

 - [ ] switch the variable names to `userManual...`

? If we could also rename the directories, that would be great. But I'm guessing that's not an option just yet.

Reply:
Regarding `new File(..., ...)`, I seem to remember that we replaced calls to `Project.file` when we already new the directory we would be relative to by `new File(buildDir, 'relative-path')`, since it is more performant.

Reply:
Happy to use `file()` where we can. I want to keep the change super-minimal here. Later I want to refactor the Gradle scripts and use Kotlin DSL. That work for you?

Reply:
@wolfs Is the improved performance worth the downsides? I don't have a very strong opinion here, and I realise this build file is unlikely to be viewed by many users.

Eric, happy to keep the changes to a minimum, particularly if the build script will be migrated to Kotlin DSL.

Reply:
It is about configuration time. So every time you do any action on the build you paid the price. I don't remember the actual numbers, maybe @oehme does?

Reply:
`Project.file` is now dirt cheap, I fixed all of its performance issues in the past releases. Of course doing a straight up `new File(foo, "bar")` is still a tiny bit faster, but it's no longer something you actively need to hunt down.